conda_smithy/lint_recipe.py

```diff
@@ -490,18 +486,17 @@ def lintify_meta_yaml(
                         lints.append(
                             "`noarch` packages can't have selectors. If "
                             "the selectors are necessary, please remove "
-                            "`noarch: {}`.".format(noarch_value)
+                            f"`noarch: {noarch_value}`."
                         )
                         break
 
     # 19: check version
-    if package_section.get("version") is not None:
-        ver = str(package_section.get("version"))
+    if (version := package_section.get("version")) is not None:
```

Comment:
Think we can simplify this further

```suggestion
    version = str(package_section.get("version", ""))
    if version:
```

Reply:
I propose to stay defensive here and crash intentionally if `version` is an empty string (as opposed to `None`). Open to split the `:=` assignment though if you think this is too fancy.

Reply:
I like John's suggestion much better.

> I propose to stay defensive here and crash intentionally if `version` is an empty string (as opposed to `None`).

That is way too subtle to go without a comment, but in any case, I don't think we need to: We should make the version check unconditional (even for `""`), which should be the best of both worlds AFAICT:
```
version = str(package_section.get("version", ""))
try:
    VersionOrder(version)
except InvalidVersionSpec as e:
    ...
```

I'm not aware of any feedstocks without a version; unless I'm overlooking a use-case here, I'd like to just get rid of that corner case completely.

Reply:
How about this solution? It takes your approach but makes it super-obvious what is going on.

0c04145

Reply:
Fine by me. There are two tests that now produce a new linter hint that needs to be handled.

Reply:
Done.